misc_scripts/update_workflow_config.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/python3
+"""
+update_workflow_config: update the GitHub Actions workflow configuration,
+either by changing the Rusttoolchain version, or changing the Fedora
+version.
+"""
+
+# isort: STDLIB
+import argparse
+
+KEY_LSRT = r"# LOWEST SUPPORTED RUST TOOLCHAIN"
+KEY_CDRT = r"# CURRENT DEVELOPMENT RUST TOOLCHAIN"
+TOOLCHAIN_CHOICES = ["lowest", "current"]
+
+KEY_LFDE = r"# LOWEST DEVELOPMENT ENVIRONMENT"
+KEY_CFDE = r"# CURRENT DEVELOPMENT ENVIRONMENT"
+KEY_NFDE = r"# NEXT DEVELOPMENT ENVIRONMENT"
+ENV_CHOICES = ["lowest", "current", "next"]
+
+
+def search_file(search_key, old_verstring, new_verstring, filename):
+    """
+    Read the file.
+
+    :param str search_key: the search key for the line to replace
+    :param str old_verstring: old version string
+    :param str new_verstring: new version string
+    :param str filename: the name of the file to read
+    """
+
+    output = []
+
+    with open(filename, "r") as file:
+        for line in file:
+
+            if search_key in line:
+                templine = line.replace(old_verstring, new_verstring)
+                if new_verstring not in templine:
+                    raise RuntimeError("Old version not in file")
+                output.append(templine)
+            else:
+                output.append(line)
+
+    return output
+
+
+def gen_parser():
+    """
+    Generate parser.
+    """
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers()
+    parser_r = subparsers.add_parser("rust")
+    parser_r.add_argument(
+        "toolchain", choices=TOOLCHAIN_CHOICES, help="the toolchain to change"
+    )
+    parser_r.add_argument("file", help="the configuration file to read")
+    parser_r.add_argument("outfile", help="the configuration file to write")
+    parser_r.add_argument("old_version", help="the old Rust version")
+    parser_r.add_argument("new_version", help="the new Rust version")
+    parser_e = subparsers.add_parser("fedora")
+    parser_e.add_argument(
+        "fedora", choices=ENV_CHOICES, help="the environment to change"
+    )
+    parser_e.add_argument("file", help="the configuration file to read")
+    parser_e.add_argument("outfile", help="the configuration file to write")
+    parser_e.add_argument("old_version", help="the old Fedora version")
+    parser_e.add_argument("new_version", help="the new Fedora version")
+
+    return parser
+
+
+def main():
+    """
+    Main method
+    """
+
+    parser = gen_parser()
+    args = parser.parse_args()
+
+    filename = args.file
+    outfilename = args.outfile
+
+    if hasattr(args, "toolchain"):
```

Comment:
I think I see what's going on here: as the program stands now, after the `search_key` variable is properly set, based on the parser called, the rest of the `main()` function processes the changes, since those instructions are identical.  But, in the future, they may not be identical.  And the set_defaults() value of `func` would determine which function needs to be called next.

Reply:
Exactly!

Reply:
OK, I created a couple of helper functions, each of which set a local `search_key` variable, based on the `func` value; right now, both call `process_file()` the same way.